app/src/main/java/de/westnordost/streetcomplete/measure/MeasureActivity.kt

```diff
@@ -0,0 +1,406 @@
+            initializeSession()
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                initRenderables()
+            }
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        if (arSceneView != null) {
+            try {
+                arSceneView?.resume()
+                binding.handMotionView.isGone = false
+                binding.trackingErrorTextView.isGone = true
+            } catch (e: CameraNotAvailableException) {
+                // without camera, we can't do anything, might as well quit
+                finish()
+            }
+        }
+    }
+
+    override fun onPause() {
+        super.onPause()
+        arSceneView?.pause()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        arSceneView?.pause()
+        arSceneView?.destroy()
+        // closing can take several seconds, should be done one background thread that outlives this activity
+        GlobalScope.launch(Dispatchers.Default) {
+            arSceneView?.session?.close()
+        }
+    }
+
+    private fun readIntent() {
+        measureMode = intent.getStringExtra(PARAM_MEASURE_MODE)?.let { MeasureMode.valueOf(it) }
+            ?: MeasureMode.BOTH
+        displayUnit = intent.getStringExtra(PARAM_DISPLAY_UNIT)?.let { Json.decodeFromString(it) }
+            ?: MeasureDisplayUnitMeter(2)
+        requestResult = intent.getBooleanExtra(PARAM_REQUEST_RESULT, false)
+    }
+
+    /* --------------------------------- Scene.OnUpdateListener --------------------------------- */
+
+    override fun onUpdate(frameTime: FrameTime) {
+        val frame = arSceneView?.arFrame ?: return
+
+        if (frame.hasFoundPlane()) {
+            binding.handMotionView.isGone = true
+        }
+
+        val textResId = frame.camera.trackingFailureReason.messageResId
+        binding.trackingErrorTextView.isGone = textResId == null
+        textResId?.let { binding.trackingErrorTextView.setText(textResId) }
+
+        if (frame.camera.trackingState == TrackingState.TRACKING) {
+            val centerX = binding.arSceneViewContainer.width / 2f
+            val centerY = binding.arSceneViewContainer.height / 2f
+            val hitResult = frame.hitPlane(centerX, centerY)
+
+            if (hitResult != null) {
+                updateCursor(hitResult)
+            }
+        }
+    }
+
+    /* ------------------------------------------ Session --------------------------------------- */
+
+    private suspend fun initializeSession() {
+        val result = createArCoreSession()
+        if (result is ArCoreSessionCreator.Success) {
+            val session = result.session
+            val config = Config(session)
+            configureSession(config)
+            session.configure(config)
+            addArSceneView(session)
+        } else if (result is ArCoreSessionCreator.Failure) {
+            val reason = result.reason
+            if (reason == ArNotAvailableReason.AR_CORE_SDK_TOO_OLD) {
```

Comment:
For me, the reason is `DEVICE_COMPATIBILITY_CHECK_FAILURE`.

And in `ArCoreSessionCreator`, `availability` is `UNKNOWN_ERROR`:
https://github.com/streetcomplete/StreetComplete/blob/13502d623f0aaf94a12efb001879a0e913c274a1/app/src/main/java/de/westnordost/streetcomplete/measure/ArCoreSessionCreator.kt#L52

After sideloading ARCore from [APKMirror](https://www.apkmirror.com/apk/google-inc/arcore/arcore-1-29-2132102-release/google-play-services-for-ar-1-29-213210293-2-android-apk-download/), the reason here is instead `DEVICE_NOT_COMPATIBLE`.

Reply:
Ah I see, makes sense. Thanks!

Reply:
Then the checks at least work as intended.